### DISCUSSION TITLE
Ability to hide entire SQL from exception log

### DISCUSSION THREAD

**Discussion by tooptoop4** [sqlalchemy/sqlalchemy#7769] [Usage Questions]
### Describe the use case

Would like some way to disable SQL being printed in errors:

https://github.com/sqlalchemy/sqlalchemy/blob/main/lib/sqlalchemy/exc.py#L505

### Databases / Backends / Drivers targeted

Redshift/Postgres

### Example Use

Don't show tokens

**Comment by CaselIT** [comment]
Hi,

You can use [`handle_error`](https://docs.sqlalchemy.org/en/14/core/events.html?highlight=event#sqlalchemy.events.ConnectionEvents.handle_error) to modify the errors.

**Comment by kblcuk** [comment]
For those who end up in this discussion and wonder "yea, but _how_?" -- here is an example from another issue: https://github.com/sqlalchemy/sqlalchemy/issues/4806#issuecomment-521501106

Though it talks about parameters substitution (which nowadays can be achieved by just passing `hideParameters = true` to sql alchemy's `create_engine`), it's quite similar with `statement` prop, just replace it with `None` or handle in whichever other way you need.
